hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/CallIdSequence.java

```diff
@@ -131,30 +139,32 @@ public long getLastCallId() {
             return longs.get(INDEX_HEAD);
         }
 
-        long getTail() {
-            return longs.get(INDEX_TAIL);
-        }
-
         @Override
         public int getMaxConcurrentInvocations() {
             return maxConcurrentInvocations;
         }
 
         @Override
         public long next(Invocation invocation) {
-            assert invocation.op.getCallId() == 0 : "callId should be null:" + invocation;
+            final Operation op = invocation.op;
+            assert !invocation.isActive() : "Invocation is already in progress:" + invocation;
```

Comment:
These are good ideas, I'll implement it that way. About the check, it's  an `assert`, not a business logic-level check. However, after all the changes it does indeed seem redundant.